### PR TITLE
Merge paradox/scala/distributed-pub-sub.md and java/distributed-pub-sub.md

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #23052

## Rendered HTML:

- https://richard-akka-doc.firebaseapp.com/scala/distributed-pub-sub.html
- https://richard-akka-doc.firebaseapp.com/java/distributed-pub-sub.html

## I split this in two commits:

- **1st commit:** Identical contents in scala/distributed-pub-sub.md and java/distributed-pub-sub.md -> **this is to take the diff both in scala/distributed-pub-sub.md and java/distributed-pub-sub.md**

  - [scala/distributed-pub-sub.md  (prev) vs. scala/distributed-pub-sub.md (after)](https://github.com/richard-imaoka/akka/commit/317d3f2ada740513d2caa5d039481193b81aded8#diff-3e2c28224a420da0acf33112716c2a52)
  - [java/distributed-pub-sub.md (prev) vs. java/distributed-pub-sub.md (after)](https://github.com/richard-imaoka/akka/commit/317d3f2ada740513d2caa5d039481193b81aded8#diff-57ceda9bbf411966ad78443cebf335ca)

- **2nd commit:** Replace java/distributed-pub-sub.md with a symlink to scala/distributed-pub-sub.md

### CODE REVIEW

File: akka-docs/src/main/paradox/scala/distributed-pub-sub.md [hunk 1]
```diff
@@ -30,9 +30,13 @@ any other node.
 There a two different modes of message delivery, explained in the sections
 [Publish](#distributed-pub-sub-publish) and [Send](#distributed-pub-sub-send) below.
 
+@@@ div { .group-scala }
+
 A more comprehensive sample is available in the
 tutorial named [Akka Clustered PubSub with Scala!](https://github.com/typesafehub/activator-akka-clustering).
```
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/scala/distributed-pub-sub.md]: No equivalent Java example exists.

As activator is deprecated, should we rather remove the reference to https://github.com/typesafehub/activator-akka-clustering?

File: akka-docs/src/main/paradox/scala/distributed-pub-sub.md [hunk 2]
```diff
@@ -43,7 +47,7 @@ Actors are registered to a named topic. This enables many subscribers on each no
 The message will be delivered to all subscribers of the topic.
 
 For efficiency the message is sent over the wire only once per node (that has a matching topic),
-and then delivered to all subscribers of the local topic representation. (See more in )
+and then delivered to all subscribers of the local topic representation.
```
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/scala/distributed-pub-sub.md]: Removing weird "(See more in )"

File: akka-docs/src/main/paradox/scala/distributed-pub-sub.md [hunk 3]
```diff
@@ -109,7 +129,7 @@ cluster aware router where the routees dynamically can register themselves.
 The message will be delivered to one recipient with a matching path, if any such
 exists in the registry. If several entries match the path because it has been registered
 on several nodes the message will be sent via the supplied `RoutingLogic` (default random)
-to one destination. The sender() of the message can specify that local affinity is preferred,
+to one destination. The sender of the message can specify that local affinity is preferred,
```
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/scala/distributed-pub-sub.md]: This is the conceptual sender of the message, not the `sender()` method of an actor.

File: akka-docs/src/main/paradox/java/distributed-pub-sub.md [hunk 4]
```diff
@@ -125,27 +148,43 @@ can explicitly remove entries with `DistributedPubSubMediator.Remove`.
 
 An example of a destination actor:
 
-@@snip [DistributedPubSubMediatorTest.java]($akka$/akka-cluster-tools/src/test/java/akka/cluster/pubsub/DistributedPubSubMediatorTest.java) { #send-destination }
+Scala
+:  @@snip [DistributedPubSubMediatorSpec.scala]($akka$/akka-cluster-tools/src/multi-jvm/scala/akka/cluster/pubsub/DistributedPubSubMediatorSpec.scala) { #send-destination }
 
-Subscriber actors can be started on several nodes in the cluster, and all will receive
-messages published to the "content" topic.
```
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/java/distributed-pub-sub.md]: Wrong sentence for Java. The Java page repeats the previous "Subscriber" sentences, but it has to be "Destination".

File: akka-docs/src/main/paradox/scala/distributed-pub-sub.md (hunk 1 above)
Comment (raboof) [inline akka-docs/src/main/paradox/scala/distributed-pub-sub.md]: As activator is deprecated indeed I'd say we don't want to link to that one anymore, but migrate any interesting samples to tech hub. In this case I think we can link to http://dev.lightbend.com/start/?group=akka&project=akka-samples-cluster-scala / http://dev.lightbend.com/start/?group=akka&project=akka-samples-cluster-java

**Review by raboof — COMMENTED** [review]
Looks good, thanks for the whitespace fixes too.

Perhaps add that techhub link and then we can merge?

File: akka-docs/src/main/paradox/scala/distributed-pub-sub.md [hunk 5]
```diff
@@ -30,8 +30,12 @@ any other node.
 There a two different modes of message delivery, explained in the sections
 [Publish](#distributed-pub-sub-publish) and [Send](#distributed-pub-sub-send) below.
 
+@@@ div { .group-scala }
+
 A more comprehensive sample is available in the
-tutorial named [Akka Clustered PubSub with Scala!](https://github.com/typesafehub/activator-akka-clustering).
+tutorial named @scala[[Akka Clustered PubSub with Scala](http://dev.lightbend.com/start/?group=akka&project=akka-samples-cluster-scala)]@java[[Akka Clustered PubSub with Java](http://dev.lightbend.com/start/?group=akka&project=akka-samples-cluster-java)].
+
+@@@
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/distributed-pub-sub.md]: Hmm, looking more closely I'm afraid the `akka-samples-cluster` sample doesn't in fact contain a pubsub example... Sorry about the churn, perhaps it's best to revert that last change and merge it with the link to Activator (until we migrate that to techhub)

Comment (richard-imaoka) [inline akka-docs/src/main/paradox/scala/distributed-pub-sub.md]: Sure, thanks for letting me know. I reverted the last commit.

**Pull request merged**